primefaces/src/main/resources/META-INF/resources/primefaces/blockui/blockui.js

```diff
@@ -53,60 +53,61 @@ PrimeFaces.widget.BlockUI = PrimeFaces.widget.BaseWidget.extend({
         this.blocker.remove();
         this.block.children('.ui-blockui-content').remove();
         $(document).off('pfAjaxSend.' + this.id + ' pfAjaxComplete.' + this.id);
-        
+
         this._super(cfg);
     },
-    
+
     /**
      * Sets up the global event listeners on the document.
      * @private
      */
     bindTriggers: function() {
         var $this = this;
-        
+
         //listen global ajax send and complete callbacks
         $(document).on('pfAjaxSend.' + this.id, function(e, xhr, settings) {
-            if(!settings || !settings.source) {
-                return;
-            }
-            var sourceId = typeof settings.source === 'string' ? settings.source : settings.source.name;
-            // we must evaluate it each time as the DOM might has been changed
-            var triggers = PrimeFaces.expressions.SearchExpressionFacade.resolveComponents($this.cfg.triggers);
-            
-            if($.inArray(sourceId, triggers) !== -1 && !$this.cfg.blocked) {
+            if($this.isXhrSourceATrigger(settings)) {
                 $this.show();
             }
-        });
-
-        $(document).on('pfAjaxComplete.' + this.id, function(e, xhr, settings) {
-            if(!settings || !settings.source) {
-                return;
-            }
-            var sourceId = typeof settings.source === 'string' ? settings.source : settings.source.name;
-            // we must evaluate it each time as the DOM might has been changed
-            var triggers = PrimeFaces.expressions.SearchExpressionFacade.resolveComponents($this.cfg.triggers);
-            
-            if($.inArray(sourceId, triggers) !== -1 && !$this.cfg.blocked) {
+        }).on('pfAjaxComplete.' + this.id, function(e, xhr, settings) {
+            if($this.isXhrSourceATrigger(settings)) {
                 $this.hide();
             }
         });
     },
-    
+
+    /**
+     * Checks whether one of component's triggers equals the source ID from the provided settings.
+     *
+     * @param {JQuery.AjaxSettings} settings containing source ID.
+     * @returns {boolean} `true` one of component's triggers equals the source ID from the provided settings.
+     */
+    isXhrSourceATrigger: function (settings) {
+        var sourceId = PrimeFaces.ajax.Utils.getSourceId(settings);
+        if (!sourceId) {
+            return false;
+        }
+        // we must evaluate it each time as the DOM might has been changed
+        var triggers = PrimeFaces.expressions.SearchExpressionFacade.resolveComponents(this.cfg.triggers);
+
+        return $.inArray(sourceId, triggers) !== -1 && !this.cfg.blocked;
```

Comment:
please fast escape in the method when !this.cfg.blocked

Reply:
Netter remove the cfg.blocked check completely from this Methode and check outside